intelmq/bots/collectors/xmpp/collector.py

```diff
@@ -21,32 +40,78 @@ def init(self):
             self.stop()
 
     def process(self):
-
         if self.xmpp is None:
-            self.xmpp = sleekxmpp.XMPPBot(self.parameters.xmpp_user + '@' + self.parameters.xmpp_server,
+            self.xmpp = XMPPClientBot(self.parameters.xmpp_user + '@'
+                                + self.parameters.xmpp_server,
                                 self.parameters.xmpp_password,
-                                self.send_message,
+                                self.parameters.xmpp_room,
+                                self.parameters.xmpp_room_nick,
+                                self.parameters.xmpp_room_password,
                                 self.logger)
             self.xmpp.connect(reattempt=True)
             self.xmpp.process()
 
+            # Add Handlers and register Plugins
+            self.xmpp.register_plugin('xep_0030')  # Service Discovery
+            self.xmpp.register_plugin('xep_0045')  # Multi-User Chat
+            self.xmpp.add_event_handler("message", self.log_message)
+            self.xmpp.add_event_handler("groupchat_message", self.log_message)
+
     def killbot(self):
         self.xmpp.disconnect(wait=True)
         self.logger.info("Disconnected")
 
+    def log_message(self, msg):
+        self.logger.debug("XMPP Received Event: %r , from %r", msg['body'],
+                        msg['from'])
+        self.logger.info("XMPP Event received")
 
-class XMPPOutputBot(sleekxmpp.ClientXMPP):
+        event = None
+        raw_msg = None
 
-    def __init__(self, jid, password, send_message, logger):
+        # Try to decode the message as json, it might be an event.
+        # if this is not possible treat it as raw.
+        # TODO: There might be additional work to decode the message
+        # as base64
+        try:
+            event = json.loads(msg['body'])
+        except:
+            self.logger.info("XMPP Could not interpret the message as json. "
+                            "Treating as Raw")
+            self.logger.debug("XMPP I'm not able to tell if it is"
+                            "base64 or not!")
+            raw_msg = msg['body']
+
+        if event:
+            self.send_message(event)
+        elif raw_msg:
+            report = Report()
+            report.add("raw", raw_msg)
+            self.send_message(report)
+
+
+class XMPPClientBot(sleekxmpp.ClientXMPP):
+
+    def __init__(self,
+                jid,
+                password,
+                room,
+                room_nick,
+                room_password,
+                logger):
         sleekxmpp.ClientXMPP.__init__(self, jid, password)
-        self.add_event_handler("session_start", self.session_start)
-        self.add_event_handler("message", self.message_logging)
+
         self.logger = logger
         self.logger.info("XMPP connected")
```

Comment:
In case of connection issues, this line will be printed anyway. Could be replaced with `initiated`.